### DISCUSSION TITLE
process.memory.usage increasing steady

### DISCUSSION THREAD

**Discussion by nd1012** [microsoft/aspire#3455] [Q&A] [answered]
I'm trying Aspire for the first time and found the `process.memory.usage` value increasing steady (n Windows 10, without serving any requests):

![image](https://github.com/dotnet/aspire/assets/10883272/50072266-8689-4a1c-a9f8-a7160bb7597f)

During writing there was a peak (marked), and after that the memory usage increases more slow.

While the Visual Studio process memory shows a different value (but stable):

![image](https://github.com/dotnet/aspire/assets/10883272/287ff055-1a54-4359-b58e-5e20b5857017)

In the Task Manager I can see the ASP.NET process memory increasing steady, too:

![image](https://github.com/dotnet/aspire/assets/10883272/a0ba9c64-46e5-467f-aff2-2963ee5a68d3)

You can find the whole project source [here (`dev` branch)](https://github.com/nd1012/wan24-AutoDiscover/tree/dev).

I run the micro service (without Aspire) on a Debian Linux, too, where the memory usage stays stable since a few hours now. Now I'm wondering - do I have a memory leak somewhere in my development environment, or does is come from Aspire? Is it a problem or normal operation?

**UPDATE**: After watching this for maybe one hour, there seemed to be some garbage collection, Maybe it was only some cache or buffer filling up data.

**Comment by davidfowl** [comment, marked as answer]
It's unlikely aspire is making your process memory leak. Instead, it's just making you aware of it (if it's even a leak that is). I recommend docs on [diagnosing a memory leak](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/debug-memory-leak) in order to get a better understanding. 

It's also likely that this could be a side effect of using [server GC](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/workstation-server-gc) being on by default in ASP.NET Core projects and uses more memory. I also recommend looking at the GC counters to get a better understanding beyond high level process memory counters.

**Reply by nd1012** [reply]
Yes, I think you're right - after watching this for maybe one hour, there seemed to be some garbage collection. Maybe it was only some cache or buffer filling up data.
